src/psyclone/tests/nemo/transformations/openacc/kernels_directive_test.py

```diff
@@ -0,0 +1,252 @@
+    to an implicit loop. '''
+    reader = FortranStringReader("program implicit_loop\n"
+                                 "real(kind=wp) :: sto_tmp(5,5)\n"
+                                 "sto_tmp(:,:) = 0.0_wp\n"
+                                 "end program implicit_loop\n")
+    code = parser(reader)
+    psy = PSyFactory(API, distributed_memory=False).create(code)
+    schedule = psy.invokes.invoke_list[0].schedule
+    acc_trans = TransInfo().get_trans_name('ACCKernelsTrans')
+    schedule, _ = acc_trans.apply(schedule.children[0:1], default_present=True)
+    gen_code = str(psy.gen)
+    assert ("  !$ACC KERNELS DEFAULT(PRESENT)\n"
+            "  sto_tmp(:, :) = 0.0_wp\n"
+            "  !$ACC END KERNELS\n" in gen_code)
+
+
+def test_multikern_if(parser):
+    ''' Check that we can include an if-block containing multiple
+    loops within a kernels region. '''
+    reader = FortranStringReader("program implicit_loop\n"
+                                 "real(kind=wp) :: sto_tmp(5)\n"
+                                 "if(do_this)then\n"
+                                 "do jk = 1, 3\n"
+                                 "  sto_tmp(jk) = jk\n"
+                                 "end do\n"
+                                 "else\n"
+                                 "do jk = 1, 5\n"
+                                 "  sto_tmp(jk) = jk\n"
+                                 "end do\n"
+                                 "end if\n"
+                                 "end program implicit_loop\n")
+    code = parser(reader)
+    psy = PSyFactory(API, distributed_memory=False).create(code)
+    schedule = psy.invokes.invoke_list[0].schedule
+    acc_trans = TransInfo().get_trans_name('ACCKernelsTrans')
+    schedule, _ = acc_trans.apply(schedule.children[0:1], default_present=True)
+    gen_code = str(psy.gen).lower()
+    assert ("!$acc kernels default(present)\n"
+            "  if (do_this) then\n"
+            "    do jk = 1, 3\n" in gen_code)
+    assert ("    end do\n"
+            "  end if\n"
+            "  !$acc end kernels\n"
+            "end program implicit_loop" in gen_code)
+
+
+def test_kernels_within_if(parser):
+    ''' Check that we can put a kernels region within an if block. '''
+    reader = FortranStringReader("program if_then\n"
+                                 "if(do_this)then\n"
+                                 "  do ji=1,jpi\n"
+                                 "    fld(ji) = 1.0\n"
+                                 "  end do\n"
+                                 "else\n"
+                                 "  fld2d(:,:) = 0.0\n"
+                                 "end if\n"
+                                 "end program if_then\n")
+    code = parser(reader)
+    psy = PSyFactory(API, distributed_memory=False).create(code)
+    schedule = psy.invokes.invoke_list[0].schedule
+    acc_trans = TransInfo().get_trans_name('ACCKernelsTrans')
+    schedule.view()
```

Comment:
remove view()s?

Reply:
Done throughout (with the exception of the test for the view method).